### PR TITLE
[Impeller] Flip the viewport on vulkan

### PR DESCRIPTION
**Pull request by iskakaushik**
Fixes: https://github.com/flutter/flutter/issues/112392

### CODE REVIEW

File: impeller/renderer/backend/vulkan/context_vk.cc [hunk 1]
```diff
@@ -27,6 +27,7 @@ namespace impeller {
 
 static std::set<std::string> kRequiredDeviceExtensions = {
     VK_KHR_SWAPCHAIN_EXTENSION_NAME,
+    VK_KHR_MAINTENANCE1_EXTENSION_NAME,  // For negative viewports.
```
Comment (chinmaygarde) [inline impeller/renderer/backend/vulkan/context_vk.cc:30]: This shouldn't be necessary if we are specifying 1.1 API requirements.

**Review by chinmaygarde — APPROVED** [review]

File: impeller/renderer/backend/vulkan/context_vk.cc (hunk 1 above)
Comment (iskakaushik) [reply impeller/renderer/backend/vulkan/context_vk.cc:30]: without this MotenVK complains :(, i think moten VK targets 1.0 by default.

Comment (chinmaygarde) [reply impeller/renderer/backend/vulkan/context_vk.cc:30]: We have to opt-into 1.1 https://github.com/flutter/engine/blob/801935148c1d5a508550b63dfe1a8b18889b5c20/impeller/renderer/backend/vulkan/context_vk.cc#L275

Comment (iskakaushik) [reply impeller/renderer/backend/vulkan/context_vk.cc]: Thanks!

File: impeller/renderer/backend/vulkan/context_vk.cc [hunk 2]
```diff
@@ -272,8 +272,8 @@ ContextVK::ContextVK(
 
   vk::ApplicationInfo application_info;
   application_info.setApplicationVersion(VK_API_VERSION_1_0);
-  application_info.setApiVersion(VK_API_VERSION_1_0);
-  application_info.setEngineVersion(VK_API_VERSION_1_0);
+  application_info.setApiVersion(VK_API_VERSION_1_1);
+  application_info.setEngineVersion(VK_API_VERSION_1_1);
```
Comment (chinmaygarde) [inline impeller/renderer/backend/vulkan/context_vk.cc:276]: The engine version can stay at 1.0. That's the "Flutter Engine" version and the implementation doesn't care about it.

**Pull request merged**